2019-10-27 Ryan Lin <ryanlin95129@gmail.com>

* timespec.h (timespec_cmp) fixed issue caused by integer overflow in which
* a very old timestamp compared with a more recent one would return a positive value
* when a negative output was expected.
* timespec_cmp now returns 1 when A > B and -1 when A < B. Behavior of timespec_cmp
* when A==B stayed the same; returns 0.

--- timespec_old.h	2019-10-26 22:54:10.382893000 -0700
+++ timespec.h	2019-10-26 23:21:59.859722000 -0700
@@ -45,8 +45,20 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if( a.tv_sec < b.tv_sec ){
+    return -1;
+  }
+  else if ( a.tv_sec > b.tv_sec ){
+    return 1;
+  }
+  else{
+    if( a.tv_nsec < b.tv_nsec )
+      return -1;
+    else if ( a.tv_nsec > b.tv_nsec )
+      return 1;
+    else
+      return 0;
+  }
 }
 
 # if ! HAVE_DECL_NANOSLEEP
